Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +15,214 @@ public static class BalanceStats
+
+            return (processedNames, moneyToCompare, isPlural);
+        }
+
+        private static (List<string> name, List<decimal>) ProcessPeopleBalance(string[] peopleAndBalances, string option)
+        {
+            var nameList = new List<string>();
+            var moneyProcessed = new List<decimal>();
+            for (var i = 0; i < peopleAndBalances.Length; i++)
+            {
+                try
+                {
+                    var moneyList = CheckData(peopleAndBalances[i], out var name);
+                    if (string.IsNullOrEmpty(name)) throw new Exception(Constants.GenericMessageError);
+
+                    switch (option)
+                    {
+                        case "max" when moneyList.Count == 0:
+                            continue;
+                        case "max":
+                            moneyProcessed.Add(moneyList.Max());
+                            break;
+                        case "min" when moneyList.Count == 0:
+                            continue;
+                        case "min":
+                            moneyProcessed.Add(CheckBalance(moneyList, option));
+                            break;
+                        case "rich":
+                        case "poor":
+                        {
+                            if (moneyList.Count < 1) continue;
+                            moneyProcessed.Add(moneyList[^1]);
+                            break;
+                        }
+                    }
+
+                    nameList.Add(name);
+                }
+                catch (Exception ex)
+                {
+                    Console.ForegroundColor = ConsoleColor.Red;
+                    Console.WriteLine(ex);
+                    Console.WriteLine($"The following data from index: {i} is not valid, skipping.");
+                    Console.WriteLine($"\"{peopleAndBalances[i]}\"");
+                    Console.ResetColor();
+                    continue;
+                }
+            }
+
+            if (nameList.Count != moneyProcessed.Count || nameList == null || moneyProcessed == null || nameList.Count == 0 || moneyProcessed.Count == 0)
+            {
+                throw new Exception($"Something went very wrong in function ProcessPeopleBalance().");
+            }
+            return (nameList, moneyProcessed);
+        }
+
+        private static decimal CheckBalance(List<decimal> moneyList, string option)
+        {
+            var balance = new List<decimal>();
+            if (moneyList.Count == 1) return moneyList[0];
+            for (var i = 0; i < moneyList.Count - 1; i++)
+            {
+                balance.Add(moneyList[i+1] - moneyList[i]);
+            }
+
+            return option switch
+            {
+                "max" => balance.Max(),
+                "min" => balance.Min(),
+                _ => 0
+            };
+        }
+
+        private static List<decimal> CheckData(string line, out string name)
```

Comment:
ambigious name. Better name would be ValidateBalancesFile.

Reply:
Im not a creative person, variables and functions names is something painful for me kkkk.